### PR TITLE
Build Paho C with Clang 7

### PR DESCRIPTION
**Pull request by uilianries**
- Applied Conan Conventions
- Updated Docker images to conanio
- Added Clang 7
- Add SHA256
- Run CMake install instead copying all artifacts
- Transformed replace_in_file in git patch

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -37,46 +39,35 @@ def configure(self):
...
         cmake.build()
 
     def package(self):
-        self.copy("edl-v10", src=self._source_subfolder, dst="licenses", keep_path=False)
-        self.copy("epl-v10", src=self._source_subfolder, dst="licenses", keep_path=False)
-        self.copy("notice.html", src=self._source_subfolder, dst="licenses", keep_path=False)
-        self.copy("*.h", dst="include", src="%s/src" % self._source_subfolder)
-        pattern_name = "*paho-mqtt3"
-        pattern_ssl = "s" if self.options.SSL else ""
-        pattern_async = "a" if self.options.asynchronous else "c"
```
Comment (danimtb) [inline conanfile.py]: I haven't check the result of this recipe but please make sure it runs well when building the asynchronous version. I am not sure what it installs but I remember it was problematic as you have to link only with one typo of the lib

Comment (uilianries) [reply conanfile.py]: excellent comment, I'll do, let's keep open this PR.

**Comment by SSE4** [comment]
@uilianries appveyor is unhappy:
```
c\1.2.0\conan\testing\build\0c3cba65cd6ee2b8280df6b968f23f14d2655965\sources\src\MQTTClient.h(389): error C2143: syntax error : missing ')' before '('
```

**Comment by uilianries** [comment]
Yeah, actually I've found a bunch of bugs yesterday, when traveling. I'll provide a new patch soon.

File: conanfile.py (hunk 1 above)
Comment (uilianries) [reply conanfile.py]: I've created a patch to build and install only what is configured by options :)

**Pull request merged**